camp/api/v1/hms_smoke/tests.py

```diff
@@ -0,0 +1,240 @@
+    def test2_ongoing_smoke(self):
+        ongoing_end = timezone.now() + timedelta(hours=1)
+        self.smoke2.end = ongoing_end
+        self.smoke2.save()
+        url = reverse("api:v1:hms_smoke:smoke-ongoing")
+        response = self.client.get(url)
+         
+        assert response.status_code == 200
+        assert len(response.json()["data"]) == 2
+        for feature in response.json()["data"]:
+            if feature['density'].lower() == 'light':
+                assert feature["id"] == str(self.smoke1.id)
+            if feature['density'].lower() == 'medium':
+                assert feature["id"] == str(self.smoke2.id)
+                
+    def test3_ongoing_smoke(self):
+        ongoing_end = timezone.now() + timedelta(hours=1)
+        self.smoke2.end = ongoing_end
+        self.smoke2.save()
+        url = reverse("api:v1:hms_smoke:smoke-ongoing")
+        final_url = f'{url}?density=medium'
+        response = self.client.get(final_url)
+        assert response.status_code == 200
+        assert len(response.json()["data"]) == 1
+        assert response.json()['data'][0]['density'] == 'medium'
+        assert response.json()['data'][0]["id"] == str(self.smoke2.id)
+        
+    def test4_ongoing_smoke(self):
+        url = reverse("api:v1:hms_smoke:smoke-ongoing")
+        final_url = f'{url}?density=heavy'
+        response = self.client.get(final_url)
+        assert response.status_code == 200
+        assert len(response.json()["data"]) == 0   
+                 
+    def test5_ongoing_smoke(self):
+        old_query = timezone.now() - timedelta(hours=1)
+        self.smoke2.timestamp = old_query
+        self.smoke2.save()
+        url = reverse("api:v1:hms_smoke:smoke-ongoing")
+        response = self.client.get(url)
+         
+        assert response.status_code == 200
+        assert len(response.json()["data"]) == 1
+        for feature in response.json()["data"]:
+            if feature['density'].lower() == 'light':
+                assert feature["id"] == str(self.smoke1.id)
+        
+        
+class Tests_DetailSmoke(TestCase):
+    """
+    setUp - create objects for testing
+            smoke1,4,5 - ongoing
+            smoke1 - light
+            smoke2,4 - medium
+            smoke3,5 - heavy
+            smoke2,5 - before latest obs
+    test1 - query for smoke by small_uuid, specifically smoke 1
+            only returned one object so not a list
+    test2 - query for a small_uuid that doesnt exist, returns 404
+    
+    """
+    def setUp(self):
+        self.smoke1 = create_smoke_objects("Light", -1, 1)
+        self.smoke2 = create_smoke_objects("MEDIUM", -1, -1)
+        self.smoke3 = create_smoke_objects("Heavy", 1, 1)
+        self.smoke4 = create_smoke_objects("MEDIUM", -1, 1)
+        self.smoke5 = create_smoke_objects("Heavy", -1, 1)
+        
+    #returns 
+    def test1_detail_smoke(self):
+        url = reverse("api:v1:hms_smoke:smoke-detail", kwargs={'smoke_id':self.smoke1.id})
```

Comment:
We want to use `instance.pk` instead of `instance.id` since `.pk` is always a shortcut to the primary key field, which is usually (but not always!) called `id`.